src/states/adult.rs

```diff
@@ -453,11 +449,22 @@ impl Adult {
 
         if self.in_authority(&signed_msg.routing_message().dst) {
             self.check_signed_message_integrity(&signed_msg)?;
-            self.routing_msg_backlog.push(signed_msg.clone());
+
+            let (msg, metadata) = signed_msg.into_parts();
+
+            match msg.content {
+                MessageContent::GenesisUpdate(info) => return self.handle_genesis_update(info),
+                _ => {
+                    let signed_msg = SignedRoutingMessage::from_parts(msg, metadata);
```

Comment:
can we just use `signed_msg.clone()` above to avoid compose from parts again?
also, with this means we can reduce some code within this branch and remain as previous that call `self.send_signed_message_to_elders(&signed_msg)?;` once at the end ?

Reply:
seem like a good idea.